internal/imagedestination/impl/compat.go

```diff
@@ -99,3 +99,9 @@ func (c *Compat) PutSignatures(ctx context.Context, signatures [][]byte, instanc
 	}
 	return c.dest.PutSignaturesWithFormat(ctx, withFormat, instanceDigest)
 }
+
+// MatchesRequiredCompression validates if compression is required by the caller while selecting a blob, if it is required
+// then function performs a match against the compression requested by the caller and compression of exisiting blob.
+func (c *Compat) MatchesRequiredCompression(options private.TryReusingBlobOptions, info types.BlobInfo) bool {
```

Comment:
- This has nothing to do with the purpose of `Compat`
- Making it a public method on an unnamed field of the `ImageDestination` makes it conceptually visible to external users via interface casts … that’s not at all a major concern (callers that do undocumented things like that are just broken), but if it is easy to avoid…

I think a stand-alone public function would be cleaner.

Reply:
Place it in `imagedestination/impl/{shared, helpers,reuse}.go`, or something like that maybe.